packages/flutter/lib/src/material/text_selection_toolbar_text_button.dart

```diff
@@ -149,11 +151,26 @@ class TextSelectionToolbarTextButton extends StatelessWidget {
     };
   }
 
+  static Color _getBackgroundColor(ColorScheme colorScheme) {
+    final bool isDefaultSurface = switch (colorScheme.brightness) {
+      Brightness.light => identical(ThemeData().colorScheme.surface, colorScheme.surface),
+      Brightness.dark => identical(ThemeData.dark().colorScheme.surface, colorScheme.surface),
+    };
+    if (!isDefaultSurface) {
+      return colorScheme.surface;
+    }
+    return switch (colorScheme.brightness) {
+      Brightness.light => _defaultBackgroundColorLight,
+      Brightness.dark => _defaultBackgroundColorDark,
+    };
+  }
+
   @override
   Widget build(BuildContext context) {
     final ColorScheme colorScheme = Theme.of(context).colorScheme;
     return TextButton(
       style: TextButton.styleFrom(
+        backgroundColor: _getBackgroundColor(colorScheme),
```

Comment:
What if you just hardcode this backgroundColor to transparent?

[In my context menu color PR](https://github.com/flutter/flutter/pull/131816/files), I set the context menu's background color to `surface` on the TextSelectionToolbar under the assumption that these TextSelectionToolbarButtons would be transparent and so look like the color behind them.  It looks like the problem in this issue is that setting textButtonTheme's backgroundColor changes the buttons to be non-transparent.

Just a hunch, let me know if it works.  If so, maybe it's a preferable solution since the logic about the color of these buttons isn't stored in two places.

Reply:
That makes sense! Changing the `backgroundColor` to `Colors.transparent` works as expected.